geth/accounts.go

```diff
@@ -153,6 +154,16 @@ func SelectAccount(address, password string) error {
 		return ErrWhisperIdentityInjectionFailure
 	}
 
+
+	swarmService, err := nodeManager.SwarmService()
+	if err != nil {
+		return err
+	}
+
+	if err := swarmService.InjectIdentity(accountKey.PrivateKey); err != nil {
```

Comment:
- it would have been better if we had sth like `swarmService.SwitchIdentity(pkey)` (or `SelectIdentity()`, or `SetIdentity`, or `SetServiceKey()`) method here, i.e. something that will not blindly copy what we have in Whisper (where identity injection makes sense). In swarm, the method forces swarm to do some changes - restart in our case - with base key/identity switched.
- again, all those if you think that proxy should exist, and you don't want to call `StartSwarm(pkey)` directly (I honestly don't see why)
- see my comments for `swarm.go`